util/patch/utils.go

```diff
@@ -60,3 +60,40 @@ func toUnstructured(obj runtime.Object) (*unstructured.Unstructured, error) {
 	}
 	return &unstructured.Unstructured{Object: rawMap}, nil
 }
+
+// unsafeFocusedUnstructured returns a shallow copy of the unstructured object given as input.
+//
+// It's not safe to modify any of the keys in the returned unstructured object, the result should be treated as read-only.
+func unsafeFocusedUnstructured(obj *unstructured.Unstructured, focus patchType, isConditionsSetter bool) *unstructured.Unstructured {
```

Comment:
Called this unsafe to make sure folks double-check how to use this function :)

Reply:
What about `unsafeFocusedCopyUnstructured` (without copy it is not clear to me what the method should do)